File: solutions/csharp/protein-translation/1/ProteinTranslation.cs
using System;

public static class ProteinTranslation
{
    public static string[] Proteins(string strand)
    {
        int numberOfCodons=(int)(strand.Length/3);
        string[] codonList = new string[numberOfCodons];
        string[] proteinList = new string[numberOfCodons];
        for(int i=0;i<numberOfCodons;i++)
        {
            codonList[i]=strand.Substring(3*i,3);    
        }
        //find stop codons
        for (int i=0;i<numberOfCodons;i++)
        {
            if(codonList[i]=="UGA" || codonList[i]=="UAG" || codonList[i]=="UAA")
            {
                i=numberOfCodons+1;
            }
            else
            {
                Console.WriteLine(codonList[i]);
                switch(codonList[i])
                {
                    case "AUG":
                        proteinList[i]="Methionine";
                        break;
                    case "UGG":
                        proteinList[i]="Tryptophan";
                        break;
                    case "UUU": case "UUC":
                        proteinList[i]="Phenylalanine";
                        break;
                    case "UUA": case "UUG":
                        proteinList[i]="Leucine";
                        break;
                    case "UAU": case "UAC":
                        proteinList[i]="Tyrosine";
                        break;
                    case "UGU": case "UGC":
                        proteinList[i]="Cysteine";
                        break;
                    case "UCU": case "UCC": case "UCA": case "UCG":
                        proteinList[i]="Serine";
                        break;
                    default:
                        //proteinList[i]="Yonofui";
                        break;
                }
            }
        }
    return Array.FindAll(proteinList, i => i != null);
    }
}

File: solutions/python/protein-translation/1/protein_translation.py
def proteins(strand):
    stop_codons=["UAA","UAG","UGA"]
    codons_list=["AUG","UUU","UUC","UUA","UUG",
                 "UCU","UCC","UCA","UCG","UAU",
                 "UAC","UGU","UGC","UGG"]
    protein_list=["Methionine", 
                  "Phenylalanine", 
                  "Phenylalanine",
                  "Leucine",
                  "Leucine",
                  "Serine",
                  "Serine",
                  "Serine",
                  "Serine",
                  "Tyrosine",
                  "Tyrosine",
                  "Cysteine",
                  "Cysteine",
                  "Tryptophan"]
    number_of_codons=int(len(strand)/3)
    proteins=[]
    stop_transcription=False
    for i in range(number_of_codons):
        codon=strand[3*i:3*i+3]
    #Find stop codons
        for j in range(3):
            if codon==stop_codons[j]:
                stop_transcription=True
                break
    #Find normal codons
        if not stop_transcription:
            for j in range(14):
                if codon==codons_list[j]:
                    proteins.append(protein_list[j])


    return proteins